### PR TITLE
change clear_team_state to not use get_team_for_anonymous_user

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** fix my implemention of `clear_team_state` to not use `get_team_for_anonymous_user`

**What changed?**

`get_team_for_anonymous_user` relies on the xmodule runtime, and we don't have access to that when `clear_user_state` is called from the LMS.

I now do the same individual user submission lookup as in `clear_user_state` and pass the submission in, and then just grab the `team_submission_uuid` off of the submission.

*Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [x] JS minified, SASS compiled
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5031](https://openedx.atlassian.net/browse/EDUCATOR-5031)

FIY: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/xblock/staff_area_mixin.py [hunk 1]
```diff
@@ -467,60 +462,57 @@ def clear_student_state(self, user_id, course_id, item_id, requesting_user_id):
...
-            logger.warning(warning_msg)
-            return
+        student_item_string = "course {} item {} user {}".format(course_id, item_id, user_id)
 
-        if user_team is None:
-            warning_msg = error_msg_base + 'but they are not on a team for course {} item {}.'.format(
-                course_id, item_id
-            )
-            logger.warning(warning_msg)
+        if not submissions:
```
Comment (Andytr1) [inline openassessment/xblock/staff_area_mixin.py:492]: Trivial nit: we are mixing string.format and %s for string formatting.

Comment (jansenk) [reply openassessment/xblock/staff_area_mixin.py:492]: Yeah, but at the top i'm building a string and then using that built string as a %s format parameter itself, idk i just wanted to try to save myself some lines
